### PR TITLE
feat: [FFM-2055]: delete flag modal improvement

### PR DESCRIPTION
**Pull request by crystal-fong**
##### Summary:

Standardising the 'confirmation' dialog when deleting a flag

##### Jira Links:
https://harness.atlassian.net/browse/FFM-2055

##### Screenshots:
Before:
image
After: 
image

You can use the following comments to re-trigger PR Checks

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier - `fix prettier`

### CODE REVIEW

**Comment by crystal-fong** [comment]
retrigger jest

**Comment by crystal-fong** [comment]
retrigger jest

File: src/modules/75-cf/components/FlagOptionsMenuButton/FlagOptionsMenuButton.tsx [hunk 1]
```diff
@@ -74,13 +75,8 @@ const FlagOptionsMenuButton = (props: FlagOptionsMenuButtonProps): ReactElement
...
     confirmText: getString('delete'),
-    message: (
-      <Text>
-        <span
-          dangerouslySetInnerHTML={{ __html: getString('cf.featureFlags.deleteFlagMessage', { name: flagData.name }) }}
-        />
-      </Text>
-    ),
+    intent: Intent.DANGER,
+    message: <String useRichText stringID="cf.featureFlags.deleteFlagMessage" vars={{ name: flagData.name }} />,
```
Comment (knagurski) [inline src/modules/75-cf/components/FlagOptionsMenuButton/FlagOptionsMenuButton.tsx:79]: I always forget this exists. We really aught to replace everywhere we're using `dangerouslySetInnerHTML`.

Comment (crystal-fong) [reply src/modules/75-cf/components/FlagOptionsMenuButton/FlagOptionsMenuButton.tsx:79]: I love yanking it out, just bc I know how to lol

File: src/modules/75-cf/hooks/useGitSync.tsx [hunk 2]
```diff
@@ -33,7 +33,6 @@ interface GitSyncFormMeta {
   gitSyncInitialValues: GitSyncFormValues
   gitSyncValidationSchema: ObjectSchema<Record<string, unknown> | undefined>
 }
-
```
[the remark this answers is not in the document]
Comment (crystal-fong) [reply src/modules/75-cf/hooks/useGitSync.tsx:36]: it was the 0 coverage thing :(

File: src/modules/75-cf/components/FlagOptionsMenuButton/FlagOptionsMenuButton.tsx [hunk 3]
```diff
@@ -7,14 +7,15 @@
 
 import React, { ReactElement } from 'react'
 import { useHistory, useParams } from 'react-router-dom'
-import { useToaster, Text } from '@wings-software/uicore'
+import { useToaster } from '@wings-software/uicore'
```
Comment (knagurski) [inline src/modules/75-cf/components/FlagOptionsMenuButton/FlagOptionsMenuButton.tsx:10]: **DON'T CHANGE ANYTHING**, but we should keep an eye out for `@wings-software/uicore` and replace with `@harness/uicore` when opportunities arise.

**Review by knagurski — APPROVED** [review]
Nice work!

File: src/modules/75-cf/hooks/useGitSync.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (knagurski) [reply src/modules/75-cf/hooks/useGitSync.tsx:36]: #annoying!

File: src/modules/75-cf/components/FlagOptionsMenuButton/FlagOptionsMenuButton.tsx (hunk 3 above)
Comment (crystal-fong) [reply src/modules/75-cf/components/FlagOptionsMenuButton/FlagOptionsMenuButton.tsx:10]: noted 👍

**Review by chrisblakely01 — APPROVED** [review]

**Pull request merged**